src/main/java/spoon/reflect/visitor/LiteralHelper.java

```diff
@@ -16,17 +17,57 @@
 	private LiteralHelper() {
 	}
 
+	private static String getBasedString(Integer value, LiteralBase base) {
+		if (base == LiteralBase.BINARY) {
+			return "0b" + Integer.toBinaryString(value);
+		} else if (base == LiteralBase.OCTAL) {
+			return "0" + Integer.toOctalString(value);
+		} else if (base == LiteralBase.HEXADECIMAL) {
+			return "0x" + Integer.toHexString(value);
+		}
+		return Integer.toString(value);
+	}
+
+	private static String getBasedString(Long value, LiteralBase base) {
+		if (base == LiteralBase.BINARY) {
+			return "0b" + Long.toBinaryString(value) + "L";
+		} else if (base == LiteralBase.OCTAL) {
+			return "0" + Long.toOctalString(value) + "L";
+		} else if (base == LiteralBase.HEXADECIMAL) {
+			return "0x" + Long.toHexString(value) + "L";
+		}
+		return Long.toString(value) + "L";
+	}
+
+	private static String getBasedString(Float value, LiteralBase base) {
+		if (base == LiteralBase.HEXADECIMAL) {
+			return Float.toHexString(value) + "F";
+		}
+		return Float.toString(value) + "F";
+	}
+
+	private static String getBasedString(Double value, LiteralBase base) {
+		if (base == LiteralBase.HEXADECIMAL) {
```

Comment:
Shouldn't we have:
```java
		if (base == LiteralBase.HEXADECIMAL) {
			return Double.toHexString(value) + "D";
		}
		return Double.toString(value) + "D";
```

Reply:
Spoon was not adding "D" suffix before this commit for doubles, so I kept this behavior. That makes sense since it is double by default.

Reply:
Ok that makes sense indeed.